internal/component/loki/source/file/file.go

```diff
@@ -329,6 +329,7 @@ func (c *Component) createReader(path string, labels model.LabelSet) (reader, er
 			c.args.Encoding,
 			pollOptions,
 			c.args.TailFromEnd,
+			c.args.LegacyPositionsFile != "",
```

Comment:
I suppose there are no unintended side effects if `LegacyPositionsFile != ""`, but there is also a non-legacy positions file present?

Reply:
Then it reads from non legacy file.